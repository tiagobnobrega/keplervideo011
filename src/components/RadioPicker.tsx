import React, { useState } from 'react';
import { StyleSheet, View } from 'react-native';
import RadioButton from './RadioButton';

export interface OptionType {
  label: string;
  value: string;
  code: string;
}

export interface RadioPickerProps {
  options: OptionType[];
  onSelect: (option: OptionType) => void;
  defaultValue?: OptionType;
}

const RadioPicker = ({ options, onSelect, defaultValue }: RadioPickerProps) => {
  const [selectedOption, setSelectedOption] = useState<OptionType | null>(
    defaultValue && Object.keys(defaultValue).length ? defaultValue : null,
  );

  const handleSelect = (option: OptionType) => {
    setSelectedOption(option);
    onSelect(option);
  };

  return (
    <View style={styles.container}>
      {options.map(option => {
        const isSelected = option.value === selectedOption?.value;
        return (
          <RadioButton
            key={option.value}
            label={option.label}
            onSelect={() => handleSelect(option)}
            selected={isSelected}
            testID={`radio-btn-${option.label}`}
          />
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'flex-start',
  },
});

export default RadioPicker;
