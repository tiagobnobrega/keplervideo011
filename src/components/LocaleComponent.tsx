import { Typography } from '@amzn/kepler-ui-components';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import {
  setCountryCode,
  settingsSelectors,
} from '../store/settings/SettingsSlice';
import { COLORS } from '../styles/Colors';
import { scaleUxToDp } from '../utils/pixelUtils';
import { localeOptions } from '../utils/translationHelper';
import RadioPicker, { OptionType } from './RadioPicker';

const LocaleComponent = () => {
  const dispatch = useDispatch();
  const countryCode = useSelector(settingsSelectors.countryCode);

  const onSelect = (option: any) => {
    dispatch(setCountryCode(option));
  };

  return (
    <View style={styles.container}>
      <Typography style={styles.label} variant="title">
        Select Country Code:
      </Typography>
      <RadioPicker
        options={localeOptions}
        onSelect={onSelect}
        defaultValue={countryCode as OptionType}
      />
    </View>
  );
};

export default LocaleComponent;

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: scaleUxToDp(25),
  },
  label: {
    color: COLORS.WHITE,
    marginRight: scaleUxToDp(20),
    fontWeight: '600',
  },
});
