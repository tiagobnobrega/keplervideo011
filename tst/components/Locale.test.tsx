import { Typography } from '@amzn/kepler-ui-components';
import { render } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';
import LocaleComponent, { styles } from '../../src/components/LocaleComponent';
import RadioPicker, { OptionType } from '../../src/components/RadioPicker';

describe('LocaleComponent', () => {
  const localeOptions: OptionType[] = [
    { label: 'Option 1', value: '1', code: 'A' },
    { label: 'Option 2', value: '2', code: 'B' },
    { label: 'Option 3', value: '3', code: 'C' },
  ];
  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('renders correctly with styles', () => {
    const tree = render(<LocaleComponent />);
    expect(tree).toMatchSnapshot();
  });
  it('Typography is present in component', async () => {
    const { UNSAFE_queryByType } = render(
      <Typography style={styles.label} variant="title" />,
    );
    const typography = UNSAFE_queryByType(Typography);
    expect(typography).toBeDefined();
  });
  it('View is present in component', async () => {
    const { UNSAFE_queryByType } = render(<View style={styles.container} />);
    const view = UNSAFE_queryByType(Typography);
    expect(view).toBeDefined();
  });
  it('renders correctly without styles', () => {
    const defaultValue: OptionType = localeOptions[0];
    const tree = render(
      <RadioPicker
        options={[]}
        onSelect={jest.fn()}
        defaultValue={defaultValue}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
