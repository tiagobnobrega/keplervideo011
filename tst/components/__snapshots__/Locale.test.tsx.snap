// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleComponent renders correctly without styles 1`] = `
<RadioPicker
  defaultValue={
    Object {
      "code": "A",
      "label": "Option 1",
      "value": "1",
    }
  }
  onSelect={[MockFunction]}
  options={Array []}
/>
`;
